src/expandableCalendar/weekCalendar.js

```diff
@@ -57,9 +57,13 @@ class WeekCalendar extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    const {updateSource, date} = this.props.context;
+    const {firstDay, context} = this.props;
+    const {updateSource, date, prevDate} = context;
 
-    if (date !== prevProps.context.date && updateSource !== UPDATE_SOURCES.WEEK_SCROLL) {
+    const weekDates = getWeekDates(date, firstDay, 'yyyy-MM-dd');
+    const isSameWeek = _.includes(weekDates, prevDate);
```

Comment:
Can we move `isSameWeek` to `dateutils` file?

Reply:
Then we can move line #188 from WeekCalendar to that as well...

Reply:
Done